src/main/java/hudson/remoting/Checksum.java

```diff
@@ -72,6 +74,7 @@ static Checksum forFile(File file) throws IOException {
     /**
      * Returns the checksum for the given URL.
      */
+    @SuppressFBWarnings(value = "URLCONNECTION_SSRF_FD", justification = "This is only used for managing the jar cache as files, not URLs.")
```

Comment:
Could you add an assertion that the URL's scheme is file or at least not http/s?

Reply:
You mean something like this
```
assert url.getProtocol().equalsIgnoreCase("file") : "Non-file URL protocol generating checksum";
```

Reply:
Yeah basically. Might want to support vfs type URLs for some servlet runtimes (JBoss/Wildfly used to do this for files in the war rather than unzipping it like Tomcat/Jetty do).

Reply:
I'm concerned here by your comment that there might be other types we might want to also include in the assert. That was why I was reluctant to add the assertion initially. I suppose it won't hurt, but since we're not certain about the total set maybe it would be better to leave it off for now.